passenger-service/src/main/java/com/modsen/passengerservice/services/PassengerService.java

```diff
@@ -0,0 +1,242 @@
+    }
+
+    public HttpStatus deletePassengerByUsername(String username) throws PassengerNotFoundException{
+        return delete(username,String.format("passenger with username: %s is not found.",
+                username),passengerRepository::findByUsername);
+    }
+
+    public HttpStatus updatePassengerByEmail(PassengerDto passengerDto)
+            throws PassengerNotFoundException{
+        return update(passengerRepository::findByEmail,
+                passengerDto::getEmail,
+                passengerDto,
+                String.format("passenger with email: %s is not found.", passengerDto.getEmail()));
+    }
+
+    public HttpStatus updatePassengerByPhone(PassengerDto passengerDto)
+            throws PassengerNotFoundException{
+        return update(
+                passengerRepository::findByPhone,
+                passengerDto::getPhone,
+                passengerDto,
+                String.format("passenger with phone: %s is not found.", passengerDto.getPhone())
+        );
+
+    }
+
+    public HttpStatus updatePassengerByUsername(PassengerDto passengerDto)
+            throws PassengerNotFoundException{
+        return update(
+                passengerRepository::findByUsername,
+                passengerDto::getUsername,
+                passengerDto,
+                String.format("passenger with username: %s is not found.", passengerDto.getUsername())
+        );
+
+    }
+
+    public ResponseEntity<Page<PassengerDto>> getLimitedList(int offset, int limit){
+        Pageable pageable = PageRequest.of(offset,limit);
+        Page<Passenger> passengerPage = passengerRepository.findAll(pageable);
+        Page<PassengerDto> result = passengerPage.map(passengerMapper::entityToDto);
+        return ResponseEntity.ok(result);
+    }
+
+    private HttpStatus update(Function<String, Optional<Passenger>> repositoryFunc,
+                                      Supplier<String> passengerDtoGetter,
+                                      PassengerDto passengerDto,
+                                      String exceptionMessage) throws PassengerNotFoundException{
+        Optional<Passenger> passenger_opt = repositoryFunc.apply(passengerDtoGetter.get());
+        if(passenger_opt.isPresent()){
+            Passenger passenger_db = passenger_opt.get();
+            if(Objects.nonNull(passengerDto.getEmail()) && !passengerDto.getEmail().isEmpty()){
+                passenger_db.setEmail(passengerDto.getEmail());
+            }
+            if(Objects.nonNull(passengerDto.getUsername()) && !passengerDto.getUsername().isEmpty()){
+                passenger_db.setUsername(passengerDto.getUsername());
+            }
+            if(Objects.nonNull(passengerDto.getPhone()) && !passengerDto.getPhone().isEmpty()){
+                passenger_db.setPhone(passengerDto.getPhone());
+            }
+            return HttpStatus.OK;
+        }else throw new PassengerNotFoundException(exceptionMessage);
```

Comment:
На будущее, пиши так
```suggestion
        } else  {
           throw new PassengerNotFoundException(exceptionMessage);
        }
```